packages/assistant/src/RPA/Assistant/library.py

```diff
@@ -0,0 +1,991 @@
+        """
+
+        # FIXME: support timeout
+
+        self._client.display_flet_window(title, height, width, on_top)
+        return self._client.results
+
+    @keyword("Ask User", tags=["dialog"])
+    def ask_user(self, timeout: int = 180, **options: Any) -> Result:
+        """Same as ``Run Dialog`` it will create a dialog from all the defined
+        elements and block until the user has handled it. It will also add
+        by default a submit and close buttons.
+
+        :param timeout: Time to wait for dialog to complete, in seconds
+        :param options: Options for the dialog
+
+        Returns a result object with all input values.
+
+        For more information about possible options for opening the dialog,
+        see the documentation for the keyword ``Run Dialog``.
+
+        Example:
+
+        .. code-block:: robotframework
+
+            Add heading     Please enter your username
+            Add text input  name=username
+            ${result}=      Ask User
+            Log    The username is: ${result.username}
+        """
+
+        # FIXME: support timeout
+
+        self.add_submit_buttons(["Submit", "Close"], "Submit")
+        self._client.display_flet_window(**options)
+        return self._client.results
+
+    @keyword("Refresh", tags=["dialog"])
+    def refresh(self):
+        """Can be used to update UI elements when adding elements while dialog is running"""
+        if self._client.page:
+            self._client.update_elements(self._client.page)
+        else:
+            raise Exception("No dialog open")
+
+    @keyword("Add Button", tags=["dialog"])
+    def add_button(
+        self, label: str, function: Union[Callable, str], *args, **kwargs
+    ) -> None:
+        """
+        ``function`` should be a python function or a Robot keyword name, args and kwargs should be valid arguments for it.
+        """
+
+        # FIXME: add some progress bar
+        # TODO: either optional or mandatory feature to block other button function calls while one is
+        # being processed (perhaps by making the button disabled during execution?)
+
+        # TODO: use logger.err and logger.debug
+        def on_click(event: ControlEvent):
+            if isinstance(function, Callable):
+                try:
+                    function(*args, **kwargs)
+                except Exception as err:
+                    print(f"on_click error with button labeled {label}")
+                    print(err)
+            else:
+                try:
+                    BuiltIn().run_keyword(function, *args, **kwargs)
+                except Exception as err:
+                    print(f"on_click error with button labeled {label}")
+                    print(err)
+
+        self._client.add_element(ElevatedButton(label, on_click=on_click))
+
+    @keyword("Add Next Ui Button", tags=["dialog"])
+    def add_next_ui_button(self, label: str, function: Union[Callable, str]):
```

Comment:
Rethink the naming here a bit. The term "next UI" doesn't mean anything to me. Maybe we should call it switching views, or something to that effect

Reply:
yes, this was already discussed needs a better naming, but we're not sure about which one yet 😄